src/Framework/ListTable/ListTable.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+namespace Give\Framework\ListTable;
+
+use Give\Framework\ListTable\Concerns\Columns;
+use Give\Framework\Support\Contracts\Arrayable;
+
+/**
+ * @unreleased
+ */
+abstract class ListTable implements Arrayable
+{
+    use Columns;
+
+    /**
+     * @var string
+     */
+    protected $locale;
+
+    /**
+     * @var array
+     */
+    private $items = [];
+
+    /**
+     * @unreleased
+     * @throws Exceptions\ColumnIdCollisionException
+     */
+    public function __construct(string $locale = '')
+    {
+        $this->locale = $locale;
+        $this->addColumns(...$this->getColumns());
+    }
+
+    /**
+     * Get table ID
+     *
+     * @unreleased
+     */
+    abstract public function id(): string;
+
+    /**
+     * Define table columns
+     *
+     * @unreleased
+     * @return ModelColumn[]
+     */
+    abstract protected function getColumns(): array;
+
+    /**
+     * Define visible table columns
```

Comment:
Let's make it clear that these are merely the _default_ visible columns.

Reply:
Resolved: https://github.com/impress-org/givewp/pull/6413/commits/cca31c0ba1b36824eaca0e6189d44c21eb1fe345